# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1496102791 25200
#      Mon May 29 17:06:31 2017 -0700
# Node ID 2fd1745398e3140c779d33641a3ea4881a1a6db1
# Parent  0e4ceda8fbcf06b63bea9b409b45075572512ed4
Implement colormap for axes objects in gnuplot (bug #48667).

* __gnuplot_draw_axes__.m: Get colormap from axis, rather than parent figure.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -460,43 +460,34 @@ function __gnuplot_draw_axes__ (h, plot_
       clim = axis_obj.clim;
     endif
 
   elseif (clim(1) == clim(2))
     clim(2)++;
   endif
 
   if (rows (parent_figure_obj.colormap) != 2)
-    ## Second pass to change color map for binary images (not sure correct)
     kids1 = kids;
     while (! isempty (kids1))
       obj = get (kids1(end));
       kids1 = kids1(1:(end-1));
 
       switch (obj.type)
-        case {"image"}
-          if (isfield (obj, "cdata") && islogical (obj.cdata))
-            parent_figure_obj.colormap = [0 0 0; 1 1 1];
-            axis_obj.clim = [0 1];
-            axis_obj.climmode = "manual";
-            break;
-          endif
-
         case "hggroup"
           ## Push group children into the kid list.
           if (isempty (kids1))
             kids1 = obj.children;
           elseif (! isempty (obj.children))
             kids1 = [kids1; obj.children];
           endif
       endswitch
     endwhile
   endif
 
-  cmap = parent_figure_obj.colormap;
+  cmap = axis_obj.colormap;
   cmap_sz = rows (cmap);
   addedcmap = [];
 
   ximg_data = {};
   ximg_data_idx = 0;
 
   if (! isempty (hlgnd))
     hlgndntrp = hlgnd.interpreter;
